Emby.Server.Implementations/IO/ManagedFileSystem.cs

```diff
@@ -20,30 +22,30 @@ public class ManagedFileSystem : IFileSystem
         private readonly bool _supportsAsyncFileStreams;
         private char[] _invalidFileNameChars;
         private readonly List<IShortcutHandler> _shortcutHandlers = new List<IShortcutHandler>();
-        private bool EnableSeparateFileAndDirectoryQueries;
+        private readonly bool EnableSeparateFileAndDirectoryQueries;
 
-        private string _tempPath;
+        private readonly string _tempPath;
 
-        private IEnvironmentInfo _environmentInfo;
-        private bool _isEnvironmentCaseInsensitive;
+        private readonly IEnvironmentInfo _environmentInfo;
+        private readonly bool _isEnvironmentCaseInsensitive;
 
-        private string _defaultDirectory;
+        private readonly string _defaultDirectory;
 
         public ManagedFileSystem(
             ILoggerFactory loggerFactory,
             IEnvironmentInfo environmentInfo,
-            string defaultDirectory,
-            string tempPath,
-            bool enableSeparateFileAndDirectoryQueries)
+            IApplicationPaths applicationPaths,
+            IConfiguration configuration)
         {
             Logger = loggerFactory.CreateLogger("FileSystem");
             _supportsAsyncFileStreams = true;
-            _tempPath = tempPath;
+            _tempPath = applicationPaths.TempDirectory;
             _environmentInfo = environmentInfo;
-            _defaultDirectory = defaultDirectory;
+            _defaultDirectory = configuration["ManagedFileSystem:DefaultDirectory"];
 
             // On Linux with mono, this needs to be true or symbolic links are ignored
```

Comment:
_thinking out loud, not necessary to implement in this PR_:
if this is for mono only, maybe we should drop that?..